MH-13655, Scheduler Message Logging

lkiesow opened the pull request:
This patch fixes a number of log statements in the message receiver for
scheduler messages, ensuring that exceptions are logged properly and
simplifying the overall logging.

This is a byproduct of the other scheduler fixes and has no real consequences except for logging.

*Work sponsored by SWITCH*

lkiesow committed:
MH-13655, Scheduler Message Logging

This patch fixes a number of log statements in the message receiver for
scheduler messages, ensuring that exceptions are logged properly and
simplifying the overall logging.

karendolan commented on modules/index-service/src/main/java/org/opencastproject/index/service/message/SchedulerMessageReceiverImpl.java:
```diff
@@ -78,92 +78,77 @@ private void executeSingle(final String mediaPackageId, final SchedulerItem sche
         } catch (SearchIndexException e) {
-          logger.error("Error retrieving the recording event from the search index: {}", getStackTrace(e));
+          logger.error("Error retrieving the recording event from the search index", e);
           return;
```
@lkiesow  since you have the mpId already, it'd be very helpful to add it to the error log so there's less digging for context when this log surfaces.

karendolan commented on modules/index-service/src/main/java/org/opencastproject/index/service/message/SchedulerMessageReceiverImpl.java:
```diff
@@ -78,92 +78,77 @@ private void executeSingle(final String mediaPackageId, final SchedulerItem sche
           EventIndexUtils.updateSeriesName(event, organization, user, getSearchIndex());
         } catch (SearchIndexException e) {
-          logger.error("Error updating the series name of the event to index: {}", getStackTrace(e));
+          logger.error("Error updating the series name of the event to index", e);
```
Please add mpId to help give context for the log.

karendolan commented:
HI @lkiesow since the mpId and sometimes  also the User is known at the point of the logging the ERROR, but is no longer available at the point of reading the ERROR from the log, it'd be really helpful to add the context at the point it is known. Otherwise it's annoying to read the ERROR in the log saying "the event failed catastrophically" without knowing which event or who or any context at all to justify the ERROR even being logged.

lkiesow commented:
Makes sense. Will add it to all the error logs.

lkiesow committed:
MH-13655, Simplify Error Logging

This patch simplifies the error logging of the scheduler message
receiver using a single catch block to log one error including all
necessary information like the event id, the received message type and a
stack trace of the actual error.

lkiesow commented:
@karendolan, done. Looking at this again, I noticed that you could actually simplify the logging quite a bit since it was basically the same catch block which was repeated over and over again. There is now one. In case an error occurs, you now get a message including the message type, the event id, and the whole stack trace.
